### PR TITLE
IDPF-147  Create the SCIM schema and map the user data

### PR DESCRIPTION
**Pull request by alsyx**
- [x] JIRA ticket referenced in title
- [x] Title is clear and concise
- [ ] Description gives any relevant detail
- [ ] Tests are up to date
- [ ] Documentation is up to date

### CODE REVIEW

File: config/urls.py [hunk 1]
```diff
@@ -18,8 +18,11 @@
 from django.contrib import admin
 from django.urls import include, path
 
+from .api import api
+
 
 urlpatterns = [
+    path("api/", api.urls),
```
Comment (0RWilliams) [inline config/urls.py:25]: Place under the ""/blank route.

File: core/schemas/scim_schema.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    formatted: str | None = None
+    streetAddress: str | None = None
+    locality: str | None = None
+    region: str | None = None
+    postalCode: str | None = None
+    country: str | None = None
+
+
+class SCIMUser(Schema):
+    schemas: List = ["urn:ietf:params:scim:schemas:core:2.0:User"]
```
Comment (0RWilliams) [inline core/schemas/scim_schema.py:44]: Missing `externalId ` field.

File: core/schemas/scim_schema.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            return [Address("work")]
+        else:
+            if "addresses" in obj:
+                return obj["addresses"]
+            else:
+                return None
+
+
+class SCIMUserOut(SCIMUser):
+    id: int = Field(alias="id")
```
Comment (0RWilliams) [inline core/schemas/scim_schema.py:102]: If id is part of the schema, include in base model, but make it optional with `= None`

Comment (alsyx) [reply core/schemas/scim_schema.py:102]: it's not part of the scim request from the external service, it must be provider's id (in this case id service) that's why i added it to the response

**Review by 0RWilliams — APPROVED** [review]

**Pull request merged**